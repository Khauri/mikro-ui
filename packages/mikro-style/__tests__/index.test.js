// Testing is pretty straightforward as this library has a single entrypoint.
// We simply load each fixture in the fixtures directory and apply snapshot testing.
// Be sure to update the snapshots when tests are changed.

import fs from 'fs';
import path from 'path';
import {expect} from 'chai';
import {fileURLToPath} from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

import {parse} from '../src/index.js';

const fixturesPath = path.join(__dirname, 'fixtures');

function* walk(dir) {
  for (const d of fs.readdirSync(dir, {withFileTypes: true})) {
    const entry = path.join(dir, d.name);
    if (d.isDirectory()) yield* walk(entry);
    else if (d.isFile()) yield entry;
  }
}

// strips the extension from the file name and formats it a bit
function testNameFromFile(file) {
  const ext = path.extname(file);
  const testName = path.relative(fixturesPath, file).replace(new RegExp(`${ext}$`), '');
  return testName.replace(/[-_]/g, ' ').replace(/\//g, ' > ');
}

for (const file of walk(fixturesPath)) {
  const testName = testNameFromFile(file);
  it(testName, async () => {
    const test = await import(file);
    expect(parse(test.default)).to.matchSnapshot();
  });
}
