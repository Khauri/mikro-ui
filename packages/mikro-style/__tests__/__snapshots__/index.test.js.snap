// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atomic > returns classes 1`] = `
Object {
  "attrs": Object {
    "sharpness": 5,
  },
  "classnames": "d-flex",
}
`;

exports[`css > returns classname 1`] = `
Object {
  "attrs": Object {},
  "classname": "1w8nbjg",
  "css": ".css-1w8nbjg {background-color:var(--mikro-colors-background);display:flex;}",
}
`;

exports[`inline > multi property styles 1`] = `
Object {
  "attrs": Object {},
  "styles": "margin-left:20px;margin-right:20px;",
}
`;

exports[`inline > no css variables 1`] = `
Object {
  "attrs": Object {},
  "styles": "color:#000000;",
}
`;

exports[`inline > returns styles string 1`] = `
Object {
  "attrs": Object {},
  "styles": "display:flex;",
}
`;

exports[`removes nullish properties 1`] = `
Object {
  "attrs": Object {},
  "styles": "",
}
`;

exports[`resolve property aliases 1`] = `
Object {
  "attrs": Object {},
  "styles": "margin-left:20px;margin-right:20px;margin-left:10px;margin-right:10px;display:flex;display:flex;",
}
`;

exports[`resolve theme value 1`] = `
Object {
  "attrs": Object {},
  "styles": "color:#000000;",
}
`;

exports[`resolve theme value recursively 1`] = `
Object {
  "attrs": Object {},
  "styles": "color:#FFF;",
}
`;
